lang/test/org/partiql/lang/planner/memorydb/InMemoryTable.kt

```diff
@@ -8,17 +9,14 @@ import org.partiql.lang.eval.ExprValueFactory
 import org.partiql.lang.eval.ExprValueType
 import java.util.TreeMap
 
-/**
- * An extermely simple in-memory table, to be used with [InMemoryDatabase].
- */
+/** An extremely simple in-memory table, to be used with [InMemoryDatabase]. */
 class InMemoryTable(
-    val name: String,
-    primaryKeyFields: List<String>,
+    val metadata: TableMetadata,
     private val valueFactory: ExprValueFactory
-) : Iterable<ExprValue> {
+) : Sequence<ExprValue> {
     private val rows = TreeMap<ExprValue, ExprValue>(DEFAULT_COMPARATOR)
```

Comment:
You'll note that `rows` on line 17 is a `TreeMap` that uses `ExprValue` as its key.  This is so that the definition equality used by `InMemoryTable` for primary keys is the same as the one that is used by PartiQL's `=` operator.